numba/tests/test_np_functions.py

```diff
@@ -375,6 +380,38 @@ def check(x_types, x_values):
         x_types = [types.complex64, types.complex128]
         check(x_types, x_values)
 
+    @unittest.skipIf(NumpyVersion(np.__version__) < '1.12.0', "Numpy bug")
+    def test_count_nonzero(self):
+
+        def arrays():
+            yield np.array([]), None
+            yield np.zeros(10), None
+            yield np.arange(10), None
+            yield np.arange(3 * 4 * 5).reshape(3, 4, 5), None
+            yield np.arange(3 * 4).reshape(3, 4), 0
+            yield np.arange(3 * 4).reshape(3, 4), 1
+
+        pyfunc = count_nonzero
+        cfunc = jit(nopython=True)(pyfunc)
+
+        for arr, axis in arrays():
+            expected = pyfunc(arr, axis)
+            got = cfunc(arr, axis)
+            self.assertPreciseEqual(expected, got)
+
+    @unittest.skipUnless(NumpyVersion(np.__version__) < '1.12.0', "Numpy bug")
+    def test_count_nonzero_exception(self):
+        pyfunc = count_nonzero
+        cfunc = jit(nopython=True)(pyfunc)
+        self.disable_leak_check()
```

Comment:
don't think this is needed, the exception is being raised at typing time, not from compiled code?